Netezos/IO/MichelineReader.cs

```diff
@@ -0,0 +1,447 @@
+﻿using Netezos.Encoding;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+
+namespace Netezos.IO
+{
+    public class MichelineReader : IDisposable
```

Comment:
The class name is misleading, because it is no related to micheline-only, but to the froging in general. I suggest name it like `ForgeReader` or `ForgingReader` and move to the `/Forging/Local` directory.